Tests/Grid/Mapping/Metadata/ManagerTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace APY\DataGridBundle\Grid\Tests\Mapping\Metadata;
+
+use APY\DataGridBundle\Grid\Mapping\Column;
+use APY\DataGridBundle\Grid\Mapping\Driver\Annotation;
+use APY\DataGridBundle\Grid\Mapping\Metadata\Manager;
+use APY\DataGridBundle\Grid\Mapping\Metadata\Metadata;
+use APY\DataGridBundle\Grid\Mapping\Metadata\DriverHeap;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Validator\Constraints\DateTime;
+use APY\DataGridBundle\Grid\Mapping\Driver\DriverInterface;
+
+class ManagerTest extends TestCase
+{
+    public function setUp()
+    {
+        $this->manager = new Manager();
+    }
+
+    public function testAddDriver()
+    {
+        $driverInterfaceMock = $this->createMock(DriverInterface::class);
+        $priority = 1;
+
+        $driverHeap = new DriverHeap();
+        $driverHeap->insert($driverInterfaceMock, $priority);
+
+        $this->manager->addDriver($driverInterfaceMock, $priority);
+
+        $this->assertAttributeEquals($driverHeap, 'drivers', $this->manager);
+    }
+
+    public function testGetDrivers()
+    {
+        $driverInterfaceMock = $this->createMock(DriverInterface::class);
+        
+        $priority = 1;
+        $driverHeap = new DriverHeap();
+        $driverHeap->insert($driverInterfaceMock, $priority);
+
+        $this->manager->addDriver($driverInterfaceMock, $priority);
+        $drivers = $this->manager->getDrivers();
+        
+        $this->assertEquals($driverHeap, $drivers);
+    }
+
+    public function testGetDriversReturnDifferentClone()
+    {
+        $driverFirstTime = $this->manager->getDrivers();
+        $driverSecondTime = $this->manager->getDrivers();
+
+        $this->assertNotSame($driverFirstTime, $driverSecondTime);
+    }
+
+    public function testWithoutDriverProvideEmptyMetadata()
+    {
+        $cols = [];
+        $mappings = [];
+        $groupBy = [];
+
+        $metadataExpected = new Metadata();
+        $metadataExpected->setFields($cols);
+        $metadataExpected->setFieldsMappings($mappings);
+        $metadataExpected->setGroupBy($groupBy);
+
+        $metadata = $this->manager->getMetadata('foo', 'bar');
+
+        $this->assertEquals($metadataExpected, $metadata);
+    }
+
+    public function testAddDriverProvideMetadata()
```

Comment:
I would change it in `testGetMetadata` (as it's the basic case that covers all `getMetadata`)